tests/test_dsp_average.py

```diff
@@ -59,24 +60,47 @@ def test_keepdims_parameters():
     assert len(signal.cshape) == len(ave2.cshape)
 
 
+@pytest.mark.parametrize('data', (
+                        pf.TimeData([[1, np.nan], [1, 2]], [1, 2]),
+                        pf.FrequencyData([[1, np.nan], [1, 2]], [1, 2])))
+def test_nan_value_averaging(data):
+    # Test average with data including NaNs.
+    if data.domain == 'time':
```

Comment:
the if case could be avoided if you pass 'time' and 'freq' as part of parametrize and get the data by `getattr(signal, 'time')` or `getattr(signal, 'freq')`